oracle-uri-proposal.md

```diff
@@ -0,0 +1,179 @@
+The event type is a kebab-case identifier which may be followed by _parameters_ enclosed in parenthesis.
+The parameters are a list of comma separated values.
+Each event type defines its parameters and together they determine the outcomes for the event.
+
+The event type will usually define how to interpret the preceding path segments.
+For example, for a `vs` event type may require that the final path segment be in the form `<team1>_<team2>` or a `weather` event type may require the last two segments to be in the form `<location>/<date>`.
+
+We explicitly disallow the oracle to communicate what the outcomes are for a particular event explicitly.
+Outcomes are always determined implicitly from the event type.
+This is done to avoid oracles independently creating events with similar but not precisely the same semantics and structure.
+For example, one oracle might have a weather event that has outcomes `sunny`, `cloudy`, `rainy` while another may have `light-rain` in addition.
+The hope is that this restriction will encourage the public collaboration on new event types and rich client support for them.
+
+Oracles should communicate to clients which specifications they are following for their event types but this is outside the scope of this document.
+Oracles who wish to control their own event type namespace partially or entirely may therefore link to their own specifications defining their own event types.
+
+### Event Fragments
+
+An event that describes a cardinal quantity like a price will have a large number of discrete outcomes.
+Normally, users will have to arrange a transactions to cover each outcome even if they only care about a small subset of outcomes.
+It is possible to optimize this by splitting up the quantity into some function of multiple sub-outcomes.
+
+For example, a price between \$0 and \$99,999 could be expressed as `10000×e + 1000×d + 100×c + b×10 + a` whereas a..e are in the range 0..10.
+This means that two users who want to make a BTC/USDT trade with boundaries of \$4,000 and \$20,000 and with increments of \$100 would only have to compute contracts for 7 + 5 + 3×5×9 = 147 possible relevant outcomes instead of the usual 100,000 outcomes. 
+
+The oracle will have to declare cryptographically its intention to release signature for each of the decimal decomposed values a..e.
+We treat all these values as related "fragments" of an event rather than separate events as they are semantically related to the same object.
+
+An event type may determine that an event is made up of several event fragments.
+When announcing an event with `n` event fragments the oracle creates only one announcement signature which signs all the cryptographic data together:
+
+`<event_id>#<fragment_nonce_1>..<fragment_nonce_n>`
+
+When attesting to a notional outcome (which is composed of the outcomes of each fragment) each fragment message is determined by the message:
+
+`<event_id>.<i>=<fragment_outcome_i>`
+
+## Example Event Types
```

Comment:
It might be nice to have test vectors for parsing these examples, and in the future when encoding is specified it might also be nice to have serialization and de-serialization test vectors along with signature point computation ones.

Reply:
Yes. I think there could be some basic ones in the URI format. Each document that specifies new event types should have have some more e.g. a `vs` parser should extract the team names.

I would like to do this soon but I want to avoid doing this inside this not very well focused document. I think each event type should be in its own document with test etc.